test/ArborX_EnableViewComparison.hpp

```diff
@@ -17,8 +17,77 @@
 #include <Kokkos_Core.hpp>
 
 #include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
 #include <boost/test/utils/is_forward_iterable.hpp>
 
+template <typename U, typename V>
+void arborxViewCheck(U const &u, V const &v, std::string const &u_name,
+                     std::string const &v_name, double tol = 0.)
+{
+  static constexpr int rank = U::rank;
+
+  bool same_dim_size = true;
+  for (int i = 0; i < rank; i++)
+  {
+    int ui = u.extent(i), vi = v.extent(i);
+    BOOST_TEST(ui == vi, u_name << " == " << v_name << " dimension " << i
+                                << " sizes"
+                                << boost::test_tools::tolerance(0.));
+    same_dim_size = (ui == vi) && same_dim_size;
+  }
+
+  if (!same_dim_size)
+    return;
+
+  auto const layout = u.layout();
+  Kokkos::Array<int, 8> index{0, 0, 0, 0, 0, 0, 0, 0};
+  auto make_index = [&]() {
+    std::stringstream sstr;
+    sstr << "(";
+    for (int i = 0; i < rank - 1; i++)
+      sstr << index[i] << ", ";
+    sstr << index[rank - 1] << ")";
+    return sstr.str();
+  };
+
+  while (index[0] != layout.dimension[0])
+  {
+    auto uval = u.access(index[0], index[1], index[2], index[3], index[4],
+                         index[5], index[6], index[7]);
+    auto vval = v.access(index[0], index[1], index[2], index[3], index[4],
+                         index[5], index[6], index[7]);
+    std::string index_str = make_index();
+
+    BOOST_TEST(uval == vval, u_name << " == " << v_name << " at " << index_str
+                                    << boost::test_tools::tolerance(tol));
```

Comment:
How does Boost.Test handle the tolerance if the value type is not the same as those values you compare (float VS double here)

Reply:
The boost [documentation](https://www.boost.org/doc/libs/1_83_0/libs/test/doc/html/boost_test/testing_tools/extended_comparison/floating_point.html#boost_test.testing_tools.extended_comparison.floating_point.type_of_the_tolerance) states

> When values of two different floating point types T and U are compared, `BOOST_TEST` uses the tolerance specified for type `boost::common_type<T, U>::type`. For instance, when setting a tolerance for mixed `float`-to-`double` comparison, the tolerance for type `double` needs to be set.`.

So that means that the tolerance type must be the same as the common type.

Reply:
I somehow thought you were using float in your matrices and that this specified tolerance would be ignored but now I see I am wrong.